erizo/src/erizo/rtp/SequenceNumberTranslator.cpp

```diff
@@ -0,0 +1,114 @@
+#include "rtp/SequenceNumberTranslator.h"
+
+#include <vector>
+
+namespace erizo {
+
+constexpr uint16_t kMaxSequenceNumberInBuffer = 511;  // = 65536 / 128 and more than 10s of audio @50fps
+constexpr uint16_t kMaxDistance = 200;
+
+DEFINE_LOGGER(SequenceNumberTranslator, "rtp.SequenceNumberTranslator");
+
+SequenceNumberTranslator::SequenceNumberTranslator()
+    : in_out_buffer_{kMaxSequenceNumberInBuffer},
+      out_in_buffer_{kMaxSequenceNumberInBuffer},
+      first_input_sequence_number_{0},
+      last_input_sequence_number_{0},
+      initial_output_sequence_number_{0},
+      initialized_{false}, reset_{false} {
+}
+
+bool SequenceNumberTranslator::lessThan(uint16_t first, uint16_t last) const {
+  uint16_t result = first - last;
+  return result > 0xF000;
+}
+
+void SequenceNumberTranslator::add(SequenceNumber sequence_number) {
+  in_out_buffer_[sequence_number.input % kMaxSequenceNumberInBuffer] = sequence_number;
+  out_in_buffer_[sequence_number.output % kMaxSequenceNumberInBuffer] = sequence_number;
+}
+
+uint16_t SequenceNumberTranslator::fill(const uint16_t &first,
+                                        const uint16_t &last) {
+  const SequenceNumber& first_sequence_number = get(first);
+  uint16_t output_sequence_number = first_sequence_number.output;
+
+  if (first_sequence_number.type != SequenceNumberType::Skip) {
+    output_sequence_number++;
+  }
+
+  for (uint16_t sequence_number = first_sequence_number.input + 1;
+       lessThan(sequence_number, last);
+       sequence_number++) {
+    add({sequence_number, output_sequence_number++, SequenceNumberType::Valid});
+  }
+
+  return output_sequence_number;
+}
+
+SequenceNumber& SequenceNumberTranslator::internalGet(uint16_t input_sequence_number) {
+  return in_out_buffer_[input_sequence_number % kMaxSequenceNumberInBuffer];
+}
+
+SequenceNumber SequenceNumberTranslator::get(uint16_t input_sequence_number) const {
+  return in_out_buffer_[input_sequence_number % kMaxSequenceNumberInBuffer];
+}
+
+SequenceNumber SequenceNumberTranslator::get(uint16_t input_sequence_number, bool skip) {
+  if (!initialized_) {
+    SequenceNumberType type = skip ? SequenceNumberType::Skip : SequenceNumberType::Valid;
+    uint16_t output_sequence_number = reset_ ? initial_output_sequence_number_ : input_sequence_number;
+    add({input_sequence_number, output_sequence_number, type});
+    last_input_sequence_number_ = input_sequence_number;
+    first_input_sequence_number_ = input_sequence_number;
+    initialized_ = true;
+    reset_ = false;
+    return internalGet(input_sequence_number);
```

Comment:
I wanted to be sure I don't copy the same object too many times, but probably the compiler knows how to do it, I'll check if there's some additional copy if I use the public version.